friendshipper/src-tauri/src/main.rs

```diff
@@ -76,346 +78,369 @@ fn force_window_to_front(window: Window) {
 
 const PORT: u16 = 8484;
 
-fn main() {
+fn main() -> Result<(), CoreError> {
     let arg = std::env::args().nth(1).unwrap_or_default();
     if arg.starts_with("friendshipper://") {
         tauri_plugin_deep_link::prepare("com.believer.friendshipper");
     } else {
         let _ = tauri_plugin_deep_link::set_identifier("com.believer.friendshipper");
     }
 
-    let (log_path, otel_reload_handle) = match logging::init(VERSION, APP_NAME) {
-        Ok(path) => path,
-        Err(e) => {
-            error!("Failed to initialize logging: {:?}", e);
-            std::process::exit(1);
-        }
-    };
-
-    match utils::process::check_for_process(APP_NAME) {
-        Ok(_) => {}
-        Err(e) => {
-            error!("Failed to check for existing process: {:?}", e);
-            std::process::exit(1);
-        }
-    }
+    if let (Some(config_file), Some(config)) = Server::initialize_app_config()? {
+        let (log_path, otel_reload_handle) = match tauri::async_runtime::block_on(async {
```

Comment:
need to use tauri's async_runtime singleton here so that the tracing runtime starts within the same tokio runtime as the main application